source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -286,6 +288,36 @@ void AuthenticatorImpl::handleGoodJwt(bool cache_hit) {
     }
   }
 
+  // Copy JWT Claim to Header
+  if (provider.claim_to_headers_size() != 0) {
+    for (const auto& header_and_claim : provider.claim_to_headers()) {
+      if (!header_and_claim.claim_name().empty() && !header_and_claim.header_name().empty()) {
+        StructUtils payload_getter(jwt_->payload_pb_);
+        const ::google::protobuf::Value* found;
+        google::jwt_verify::StructUtils::FindResult claim_value =
+            payload_getter.GetValue(header_and_claim.claim_name(),found);
+        if (claim_value == StructUtils::OK) {
+          if (found->kind_case() == google::protobuf::Value::kStringValue) {
+            headers_->addCopy(Http::LowerCaseString(header_and_claim.header_name()),
+                              found->string_value());
+          } else if (found->kind_case() == google::protobuf::Value::kNumberValue) {
+            headers_->addCopy(Http::LowerCaseString(header_and_claim.header_name()),
+                              std::to_string(static_cast<uint64_t>(found->number_value())));
```

Comment:
Why cast  `double` to uint64_t?   This jwt_authn filter can be used by others,  can we just do 
std::to_string<double>

Reply:
The reason to convert this uint64_t becuase our method `GetValue()` can't return type `uint64`. It just returns this https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/struct.pb.h#L350 which is of type double. And Generally JWT Claims are not double or float, so it's safe to convert them to Int

Reply:
ok, then.  But please check its range before static_cast<uint64>, same as the jwt_verify_lib.

Reply:
sure